### PR TITLE
Added support for secret file credentials, bumped gojenkins to latest…

### PR DESCRIPTION
**Pull request by jaceq**
… and made needed adjustments.

Signed-off-by: Jacek Kikiewicz <public@kikiewicz.com>

# Dependencies

gojenkins updated to latest (already done as part of this commit)

# What Is Changing

I added support for secret file credential type, as part of this I had to update gojenkins dep to latest version because FileCredential wasn't supported by it before ( https://github.com/bndr/gojenkins/pull/233 ). Given there were some other changes in gojenkins in between I had to make adjustments all over the place to make it work with latest gojenkins.
Example and doc is added.
Test is added.
ACC tests are green locally.

# Test Steps

```sh
make testacc
```

### CODE REVIEW

**Comment by jaceq** [comment]
Hi @taiidani 
Regarding test coverage: I basically (once again ;) ) copied secret_text and adjusted it, same with tests. Test coverage is pretty much the same as for other credentials. Please review and let me know what you think.

Regards

**Comment by jaceq** [comment]
Hi @taiidani any update here, by any chance? I am waiting for this with some further work (I also want to add SSHkey credential support, that will however not require gojenkins changes / update)

File: jenkins/resource_jenkins_credential_secret_file_test.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckJenkinsCredentialSecretFileDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: `
+				resource jenkins_credential_secret_file foo {
+				  name = "test-secret-file"
+				  filename = "secret.txt"             
+				  secretbytes = "VGhpcyBpcyBhIHRlc3Qu"
```
Comment (taiidani) [inline jenkins/resource_jenkins_credential_secret_file_test.go:27]: I'd allow this, but consider a test strategy more reflective of the examples in the docs.

e.g. use [ t's TempDir](https://pkg.go.dev/testing#T.TempDir) and add a dummy file to it, then use Terraform's `base64encode` on the file contents.

File: jenkins/resource_jenkins_credential_secret_file_test.go [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+				),
+			},
+			{
+				// Update by adding description
+				Config: `
+				resource jenkins_credential_secret_file foo {
+				  name = "test-secret-file"
+				  description = "new-description"
+                                  filename = "secret.txt"             
+                                  secretbytes = "VGhpcyBpcyBhIHRlc3Qu"
```
Comment (taiidani) [inline jenkins/resource_jenkins_credential_secret_file_test.go:41]: Maybe update the contents of the `secretbytes` property to prove out that the provider supports update operations for the value?

**Review by taiidani — CHANGES_REQUESTED** [review]
Adding a few value-add suggestions, mostly around the tests. Note that the only items I consider blockers for a merge are the typos in the documentation and the `go fmt` ask. The others are nice-to-haves.

Really excited to see that the downstream library added context support!

File: docs/resources/credential_secret_file.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+The following arguments are supported:
+
+* `name` - (Required) The name of the credentials being created. This maps to the ID property within Jenkins, and cannot be changed once set.
+* `domain` - (Optional) The domain store to place the credentials into. If not set will default to the global credentials store.
+* `folder` - (Optional) The folder namespace to store the credentials in. If not set will default to global Jenkins credentials.
+* `scope` - (Optional) The visibility of the credentials to Jenkins agents. This must be set to either "GLOBAL" or "SYSTEM". If not set will default to "GLOBAL".
+* `description` - (Optional) A human readable description of the credentials being stored.
+* `filename` - (Required) The secret file filename on jenkins server side.
+* `secretbyes` - (Required) The secret file, base64 encoded content. I can be sourced directly from local file with filebase64(path) TF function or given directly.
```
[the remark this answers is not in the document]
Comment (jaceq) [reply docs/resources/credential_secret_file.md]: done

File: jenkins/resource_jenkins_credential_secret_file.go [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+func resourceJenkinsCredentialSecretFileUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cm := meta.(jenkinsClient).Credentials()
+	cm.Folder = formatFolderName(d.Get("folder").(string))
+
+	domain := d.Get("domain").(string)
+	cred := jenkins.FileCredentials{
+		ID:          d.Get("name").(string),
+		Scope:       d.Get("scope").(string),
+		Description: d.Get("description").(string),
+                Filename:    d.Get("filename").(string),
```
[the remark this answers is not in the document]
Comment (jaceq) [reply jenkins/resource_jenkins_credential_secret_file.go]: done

File: jenkins/resource_jenkins_credential_secret_file_test.go (hunk 1 above)
Comment (jaceq) [reply jenkins/resource_jenkins_credential_secret_file_test.go]: done

File: jenkins/resource_jenkins_credential_secret_file_test.go (hunk 2 above)
Comment (jaceq) [reply jenkins/resource_jenkins_credential_secret_file_test.go]: done

**Comment by jaceq** [comment]
Hi @taiidani ,

Thank you for review and your comments.
I have made requested changes, (including nice to have ones ;) ) please have another look when you get a chance.

Thanks!

**Review by taiidani — APPROVED** [review]

**Comment by taiidani** [comment]
@jaceq Thanks again for your contributions!

**Pull request merged**